bindings/dotnetcore/src/dotnetcore.cpp

```diff
@@ -34,13 +34,13 @@ typedef void(*PGatewayDestroyDelegate)(unsigned int moduleIdManaged);
 
 typedef void(*PGatewayStartDelegate)(unsigned int moduleIdManaged);
 
-static PGatewayCreateDelegate GatewayCreateDelegate;
```

Comment:
Yep. It was for Unit Test. Not sure what you mean by 2 modules with different delegates. The delegates goes to Microsoft.Azure.IoT.Devices_Gateway, they are the same for multiple modules. 
That said, we have end2End test (which has just 1 module). 
But yes, I've run the sample which has 2 dotnet core module and they are working with this change.